docs/DeveloperGuide.md

```diff
@@ -1,29 +1,204 @@
+
+The `TextUi`class performs the following functions:
+1. Reading the user input (Empty inputs will be ignored)
+
+    method: `getUserInput()`
+2. Printing messages generated by the app to the terminal
+
+    method: `printBanner()`, `printMessage()`
+
+### Utils Component (Darren)
+
+![UtilsParserClassDiagram](uml/images/UtilsParserClassDiagram.png)    
+The Utils Component consists of the `Parser` class that will handle the parsing and preparing of commands within the DinerDirector application.  
+
+The `Parser` class performs the following functions:  
+* Parse the command given the user input and extracts out the necessary information related to the command.
+* Returns the appropriate Command Class based on the parsed input.
+
+![UtilsStorageClassDiagram](uml/images/UtilsStorageClassDiagram.png)  
+The Utils Component also consists of the `Storage` class and the individual `XYZStorage` that will handle storage related operations within the DinerDirector application.  
+
+The `Storage` class performs the following functions:
+* Create a directory in the root folder of where the application is running.
+
+The `XYZStorage` class performs the following functions:
+* Read and load data from the file.
+* Write to the file if any changes occur to the list.
+
+## Implementation
+### Parsing Feature (Darren)
+
+![ParserSquenceDiagram](./uml/images/ParserSequenceDiagram.png)  
+How the parsing works:
+1. The `Parser()` class will be called to create a new instance of `Parser`.
+2. Afterwards, when the `parseCommand()` method is called from `DinerDirector` class, the `parseCommand()` will split the given userInput first.
+2. With the `userInputSplit[]`, the `0` index will be extracted out. That will be used as identification for the command the user typed in.
+3. The `commandWord` will be used in the switch statement to select the appropriate command. returning `IncorrectCommand` class is the default behavior.
+4. If the `commandWord` is valid, it will run the appropriate `prepareXYZCommand()`.
+5. Each of the individual `prepareXYZCommand()` will take in the userInput without the command portion. The variable is named `userInputNoCommand`. The `prepareXYZCommand()` will check the userInput to see if all the appropriate values are added, and return `XYZCommand` class if the values are correct. `prepareXXXCommand()` will return `IncorrectCommand` class if there are some missing values or inappropriate values.
+
+### Meeting Feature (PeiHao)
+The 'Meeting' Feature allows users to add a meeting, delete a meeting, find a meeting and print all the meetings.
+The 'Meeting' class in the entity package shows the attributes that a meeting object has: time and issue, both are of String type.
+An ArrayList of meetings is initialized in the MeetingManager. Three methods that implement the meetings are also inside.
+- addMeeting: Add a meeting to the meeting list.
+  ![](./uml/AddMeetingCommandSequenceDiagram-Add_Meeting_Sequence_Diagram.png)
```

Comment:
<img width="358" alt="image" src="https://user-images.githubusercontent.com/88129105/228765262-a2fb4860-d542-4576-a818-344811f6c8f9.png">
Are these frames for parallel paths? If so, are they supposed to be connected as a single frame separated by a dotted line?